ui/lib/ldap/addon/components/page/role/create-and-edit.hbs

```diff
@@ -0,0 +1,76 @@
+<PageHeader as |p|>
+  <p.top>
+    <Page::Breadcrumbs @breadcrumbs={{@breadcrumbs}} />
+  </p.top>
+  <p.levelLeft>
+    <h1 class="title is-3">
+      {{if @model.isNew "Create Role" "Edit Role"}}
+    </h1>
+  </p.levelLeft>
+</PageHeader>
+
+<hr class="is-marginless has-background-gray-200" />
+
+<form {{on "submit" (perform this.save)}} class="has-top-margin-m">
+  <MessageError @errorMessage={{this.error}} />
+
+  <label class="is-label">
+    Role type
+  </label>
+  <Hds::Form::RadioCard::Group @name="role type options" class="has-bottom-margin-m" as |RadioGroup|>
+    {{#each this.roleTypeOptions as |option|}}
+      <RadioGroup.RadioCard
+        @checked={{eq option.value @model.type}}
+        @disabled={{not @model.isNew}}
+        {{on "change" (fn (mut @model.type) option.value)}}
+        data-test-radio-card={{option.value}}
+        as |Card|
+      >
+        <Card.Icon @name={{option.icon}} />
+        <Card.Label>{{option.title}}</Card.Label>
+        <Card.Description>{{option.description}}</Card.Description>
+      </RadioGroup.RadioCard>
+    {{/each}}
+  </Hds::Form::RadioCard::Group>
+
+  {{#each @model.formFields as |field|}}
+    {{! display section heading ahead of ldif fields }}
+    {{#if field.options.sectionHeading}}
```

Comment:
I should have responded sooner because now I can't remember exactly but there definitely was a reason because I originally tried using form groups 😁!

If I recall now it's because we need to dynamically generate the form fields based on the role type (static or dyanmic), so I am not passing anything into the `withFormFields` decorator and relying on the `allFields` array to filter through when the type is changed.